libslirp static link error

### Description / Steps to reproduce the issue

After libslirp is updated to 4.8.0, I get static link error using libslirp.

Edit: Fixed according to @MehdiChinoune 's information.

### Expected behavior

Link successfully as it did before.

### Actual behavior

```
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x97): undefined reference to `__imp_slirp_input'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x69f): undefined reference to `__imp_slirp_cleanup'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x975): undefined reference to `__imp_slirp_remove_hostfwd'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x10a6): undefined reference to `__imp_slirp_pollfds_fill'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x1109): undefined reference to `__imp_slirp_pollfds_poll'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x1a6e): undefined reference to `__imp_slirp_add_hostfwd'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x1c63): undefined reference to `__imp_slirp_version_string'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: misc/libmisc.a(ethernet_slirp.o):ethernet_slirp:(.text+0x213f): undefined reference to `__imp_slirp_new'
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

github runner `windows-latest`

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

_No response_

--- a/mingw-w64-libslirp/PKGBUILD
+++ b/mingw-w64-libslirp/PKGBUILD
@@ -4,7 +4,7 @@ _realname=libslirp
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=4.8.0
-pkgrel=1
+pkgrel=2
 pkgdesc="General purpose TCP-IP emulator (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw64' 'mingw32' 'ucrt64' 'clang64' 'clangarm64')
@@ -18,13 +18,16 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-meson"
              "${MINGW_PACKAGE_PREFIX}-ninja"
              "${MINGW_PACKAGE_PREFIX}-pkgconf")
-source=("https://gitlab.freedesktop.org/slirp/${_realname}/-/archive/v${pkgver}/${_realname}-v${pkgver}.tar.bz2")
+source=("https://gitlab.freedesktop.org/slirp/${_realname}/-/archive/v${pkgver}/${_realname}-v${pkgver}.tar.bz2"
+        "https://gitlab.freedesktop.org/slirp/libslirp/-/commit/84361923047f867561efc0184a4af9c66c8ba852.patch")
 noextract=("${_realname}-v${pkgver}.tar.bz2")
-sha256sums=('1c2e3d30e41a055ed41f9d0b3234d0d143bc28f5ad84bb101413f830844df757')
+sha256sums=('1c2e3d30e41a055ed41f9d0b3234d0d143bc28f5ad84bb101413f830844df757'
+            'c8015e2e6bc0fd2a5aacc61424ab18b861320577b55ea7b15780b8e93f4b8a49')
 
 prepare() {
   bsdtar -xf "${_realname}-v${pkgver}.tar.bz2" 2>/dev/null || bsdtar -xf "${_realname}-v${pkgver}.tar.bz2"
   cd "${_realname}-v${pkgver}"
+  patch -p1 -i "${srcdir}/84361923047f867561efc0184a4af9c66c8ba852.patch"
 }
 
 build() {
@@ -61,6 +64,8 @@ build() {
 package() {
   ${MINGW_PREFIX}/bin/meson.exe install -C "build-${MSYSTEM}-static" --destdir "${pkgdir}"
   ${MINGW_PREFIX}/bin/meson.exe install -C "build-${MSYSTEM}-shared" --destdir "${pkgdir}"
+
+  echo "Cflags.private: -DLIBSLIRP_STATIC" >> "${pkgdir}${MINGW_PREFIX}"/lib/pkgconfig/slirp.pc
 
   install -Dm644 "${srcdir}/${_realname}-v${pkgver}/COPYRIGHT" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/COPYRIGHT"
 }

